.gitlab-ci.yml

```diff
@@ -1,23 +1,24 @@
 image: node:14.15.1
 
 cache:
+  key: "$CI_COMMIT_REF_NAME"
   paths:
-    - .yarn/
     - node_modules/
-    - packages/*/node_modules/*
-    - server/static/node_modules/
+    - packages/*/node_modules/
     - .jestcache/
     - .eslintcache
     - .stylelintcache
 
-before_script:
-  - yarn install --cache-folder .yarn --frozen-lockfile
-  - lerna bootstrap --concurrency=1 --mutex network --force-local
-
 stages:
+  - setup
   - quality
   - build
 
+setup:
+  stage: setup
+  script:
+    - yarn install --frozen-lockfile
+    - yarn lernabootstrap:ci
```

Comment:
I'm not sure the setup stage does what you want. Before you had that in a before_script section. The way it is setup now, it'll run `yarn install --frozen-lockfile` and `yarn lernabootstrap:ci` in a separate docker container rather then before everything else. Is this intended?